### PR TITLE
feature: add resize interface

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: Michael Wan <zirenwan@gmail.com>

### Ⅰ. Describe what this PR did
add a resize interface for pouch

### Ⅱ. Does this pull request fix one issue?
fixes https://github.com/alibaba/pouch/issues/866

### CODE REVIEW

File: apis/swagger.yml [hunk 1]
```diff
@@ -2447,6 +2470,15 @@ definitions:
         additionalProperties:
           type: "string"
 
+  ResizeOptions:
```
Comment (allencloud) [inline apis/swagger.yml]: If this is only used in daemon side, I do not think we should list this in swagger.yml.

**Comment by allencloud** [comment]
Since https://github.com/alibaba/pouch/pull/881 has been merged, please rebase to the latest branch and push back again. Thanks. @HusterWan

File: apis/swagger.yml (hunk 1 above)
Comment (HusterWan) [reply apis/swagger.yml]: No, this struct is used by both

**Comment by sunyuan3** [comment]
ping @HusterWan
A error leads to the CI failure. 
Step to reproduce:

```
git clone -b zr/add-resize-interface https://github.com/HusterWan/pouch.git
cd pouch
sudo make -e test
```
Log is as follow.
```
root@p0:/go/src/github.com/alibaba/pouch# docker run --rm -v /go/src/github.com/alibaba/pouch:/go/src/github.com/alibaba/pouch pouch:test bash -c 'make build'
GOOS=linux go build -o pouchd
# github.com/alibaba/pouch/daemon/mgr
daemon/mgr/container.go:498:9: cannot use mgr (type *ContainerManager) as type ContainerMgr in field value:
	*ContainerManager does not implement ContainerMgr (wrong type for Resize method)
		have Resize(string, "github.com/alibaba/pouch/apis/types".ResizeOptions) error
		want Resize("context".Context, string, "github.com/alibaba/pouch/apis/types".ResizeOptions) error
Makefile:23: recipe for target 'server' failed
make: *** [server] Error 2
```

File: apis/swagger.yml [hunk 2]
```diff
@@ -500,6 +500,29 @@ paths:
...
+          description: "height of the tty"
+          type: "string"
+        - name: "width"
+          in: "query"
+          description: "width of the tty"
+          type: "string"
+      responses:
+        200:
+          description: "no error"
+        404:
```
Comment (allencloud) [inline apis/swagger.yml]: You returned 400(StatusBadRequest), and I am afraid you should add this in your swagger.yml to keep consistent.

File: apis/swagger.yml (hunk 1 above)
Comment (allencloud) [reply apis/swagger.yml]: I see. I found that you are not using this `ResizeOptions ` to communicate between client side and daemon side. You indeed use `ResizeOptions ` in package client, but when sending request, you split this struct into two separate parameters:

```
query.Set("h", strconv.Itoa(int(opts.Height)))
query.Set("w", strconv.Itoa(int(opts.Width)))
```

So in daemon side, you use both two parameters above to construct this struct `ResizeOptions` again.

I am afraid that the way is not the same as other struct used by both client and daemon side.

Correct me if I missed something, please.

**Comment by allencloud** [comment]
LGTM

**Pull request merged**